File: c/3427.c
int subarraySum(int *nums, int numsSize)
{
  int sub_sum = 0;

  for (int i = 0; i < numsSize; i++)
  {
    int start = i - nums[i] > 0 ? i - nums[i] : 0;

    for (int j = start; j < i + 1; j++)
    {
      sub_sum += nums[j];
    }
  }

  return sub_sum;
}

File: python/3427.py
from typing import List


class Solution:
    def subarraySum(self, nums: List[int]) -> int:
        sub_sum: int = 0

        for i in range(len(nums)):
            start: int = max(0, i - nums[i])

            sub_sum += sum(nums[start : i + 1])

        return sub_sum
